New String.TrimEnd overload breaks MSBuild `CoerceArguments` on netcoreapp2.0

CoreFX has added a new String.TrimEnd overload with https://github.com/dotnet/corefx/pull/15334.

This is breaking MSBuild running on netcoreapp2.0 when building a project that contains logic like the following [taken from dotnet/sdk](https://github.com/dotnet/sdk/blob/9dc299b606ba76d1860700ed27016920afdd25e2/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.TargetFrameworkInference.targets#L51):

```
<_ShortFrameworkIdentifier>$(TargetFramework.TrimEnd('.0123456789'))</_ShortFrameworkIdentifier>
```

I get the following error:

```
Microsoft.NET.TargetFrameworkInference.targets(51,4): error MSB4184: The expression ""netcoreapp2.0".TrimEnd(.0123456789)" cannot be evaluated. String must be exactly one character long. 
```

I've debugged the error down to [this logic in MSBuild](https://github.com/Microsoft/msbuild/blob/7a49764a4cf5a5aef2e16f5900edf61bd295dcc1/src/XMakeBuildEngine/Evaluation/Expander.cs#L3607-L3611):

```
                        else
                        {
                            // change the type of the final unescaped string into the destination
                            coercedArguments[n] = Convert.ChangeType(args[n], parameters[n].ParameterType, CultureInfo.InvariantCulture);
                        }
                    }
                }
                catch (InvalidCastException)
                {
                    // The coercion failed therefore we return null
                    return null;
```

The issue is `Convert.ChangeType` is throwing a `FormatException`, [just like it is documented to do](https://msdn.microsoft.com/en-us/library/ms130977(v=vs.110).aspx).  But this code only catches `InvalidCastException`.  Instead, since trying to convert a string to a single char fails, this method overload should be skipped and keep searching until it finds the overload that accepts `char[]`.

This is blocking the CLI "2.0" work from taking new builds of the .NET Core runtime.

--- a/src/XMakeBuildEngine/Evaluation/Expander.cs
+++ b/src/XMakeBuildEngine/Evaluation/Expander.cs
@@ -3608,9 +3608,13 @@ namespace Microsoft.Build.Evaluation
                         }
                     }
                 }
+                // The coercion failed therefore we return null
                 catch (InvalidCastException)
                 {
-                    // The coercion failed therefore we return null
+                    return null;
+                }
+                catch (FormatException)
+                {
                     return null;
                 }
 

